web-jehlomat/src/screens/Nalezy/Components/FilterByRange.tsx

```diff
@@ -1,25 +1,48 @@
-import React, { ChangeEvent, FunctionComponent, useEffect, useState } from 'react';
-import dayjs, { Dayjs } from 'dayjs';
+import React, { ChangeEvent, FunctionComponent, useCallback, useEffect, useState } from 'react';
+import { useSetRecoilState } from 'recoil';
+import dayjs from 'dayjs';
+import { RangeKind } from 'screens/Nalezy/types/RangeKind';
+import { Range as DateRange } from 'screens/Nalezy/types/Range';
+import { Filtering } from 'screens/Nalezy/types/Filtering';
+import { filteringState } from 'screens/Nalezy/store';
 import { Filter, Range } from './Filter';
-import Select from './Select';
 import { DatePicker } from './Input';
-import { RangeKind } from '../types/RangeKind';
-
-interface FilterPeriodProps {
-    onFilter: (kind: RangeKind, from: Dayjs | null, to: Dayjs | null) => void;
-    onReset: () => void;
-}
+import Select from './Select';
 
-const FilterByRange: FunctionComponent<FilterPeriodProps> = ({ onFilter, onReset }) => {
+const FilterByRange: FunctionComponent = () => {
     const [kind, setKind] = useState<RangeKind | ''>('');
     const [from, setFrom] = useState<string>('');
     const [to, setTo] = useState<string>('');
+    const setFilter = useSetRecoilState(filteringState);
+
+    const filtering = useCallback((kind: RangeKind, range: DateRange) => {
+        setFilter((state: Filtering) => {
+            const filter = { ...state };
+            delete filter.createdAt;
+            delete filter.demolishedAt;
+
+            if (kind === 'DEMOLISH') return { ...filter, demolishedAt: range };
```

Comment:
can we make this (kind) Enum and compare against it? This is anarchy 😂

Reply:
In this case I prefer union type in this case.

Reply:
why do you prefer non-typesafety?

Reply:
You are comparing variable to a string not to a constant in here. You miss a character somewhere and you are fucked. Where is safety in that?

Reply:
Unless you're writing in Notepad, an IDE start yelling, Typescript will throw an error during transpilation, hot reload shows that error in browser and even if you try to build app, it'll fail. That's enough, isn't it?